_integration-tests/utils/backoff/backoff.go

```diff
@@ -0,0 +1,105 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package backoff provides utilities to retry operations when encoutering
+// transient errors. It is used by integration tests to allow for testcontainers
+// enough time to become fully ready, avoiding tests flaking because the CI
+// resources are constrained enough to cause containers to not be "immediately"
+// ready to serve traffic.
+package backoff
```

Comment:
I wonder if we could use https://github.com/cenkalti/backoff instead of implementing this ourselves 🤔